### PR TITLE
Replace COPYRIGHT file from the root with NOTICE file

### PR DESCRIPTION
**Pull request by yeputons**
The primary reason: make GitHub detect that we use Apache License 2.0
They do it via https://github.com/licensee/licensee Ruby library (gem).

Our COPYRIGHT file contains a part of the Apache License, which should
be added to a source file, not the license or copyright information itself,
which confuses the library.

Losing a note about vendor/postgres having a different license should be fine,
as it is a clearly separate git repository with its own COPYRIGHT file.

### CODE REVIEW

**Comment by hlinnaka** [comment]
Hmm, the "How to apply the Apache license to your work" section at https://www.apache.org/licenses/LICENSE-2.0 says:
> Include a copy of the Apache License, typically in a file called LICENSE, in your work, and consider also including a NOTICE file that references the License.

Should we move the notice about vendor/postgres to a file called NOTICE?

**Comment by yeputons** [comment]
Good idea corrected the commit. The license should still be detected correctly.

File: NOTICE [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Neon
+Copyright 2022 Neon Inc.
```
Comment (yeputons) [inline NOTICE:2]: I believe that's the name of our legal entity?

File: NOTICE [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+The PostgreSQL submodule in vendor/postgres is licensed under the
+PostgreSQL license. See vendor/postgres/COPYRIGHT.
```
Comment (hlinnaka) [inline NOTICE:5]: Let's have a blurp here, saying something like "this is licensed under the Apache license". Best to copy-paste that from somewhere, to use some common phrasing of that.

**Review by hlinnaka — APPROVED** [review]

File: NOTICE (hunk 2 above)
Comment (yeputons) [reply NOTICE:5]: Hmm, I did not see anything like this in Apache's examples like https://infra.apache.org/licensing-howto.html

Feel free to alter as you deem necessary and merge. I believe the `licensee` library does not care about `NOTICE` at all.

**Comment by yeputons** [comment]
@hlinnaka As we're trending at https://github.com/trending/rust, I decided to go ahead and merge this PR right now. I didn't really understand what kind of blurps you're referring to, and the current state seems in line with Apache's recommendations to me.

I don't have any strong opinion on the content of the `NOTICE` file, though.

**Pull request merged**